internal/clusterconfig/lvmconfig.go

```diff
@@ -94,17 +96,48 @@ func (r *UpgradeClusterConfigGather) fetchLocalVolumes(ctx context.Context, mani
 		return fmt.Errorf("failed to list localvolumes: %w", err)
 	}
 
+	var scNameSet = make(map[string]bool)
 	for _, lv := range lvsList.Items {
 		// Unset uid and resource version
 		lv.SetUID("")
 		lv.SetResourceVersion("")
 
-		filePath := filepath.Join(manifestsDir, lv.GetName()+"_"+lv.GetNamespace()+".json")
+		lvFileName := fmt.Sprintf("%s_%s_%s.json", lv.GetKind(), lv.GetName(), lv.GetNamespace())
+		filePath := filepath.Join(manifestsDir, lvFileName)
 		r.Log.Info("Writing LocalVolume to file", "path", filePath)
 		err := utils.MarshalToFile(lv.Object, filePath)
 		if err != nil {
 			return fmt.Errorf("failed to write localvolume to %s: %w", filePath, err)
 		}
+
+		// Get the associated storage classes names
+		lvSpec := lv.Object["spec"].(map[string]any)
+		if scDevices, exists := lvSpec["storageClassDevices"].([]any); exists {
+			for _, sc := range scDevices {
+				if scName, exists := sc.(map[string]any)["storageClassName"].(string); exists {
+					scNameSet[scName] = true
+				}
+			}
+		}
+	}
+
```

Comment:
I'd add a quick check `if len(scNameSet) == 0` then return before the `for` below, but for your consideration

Reply:
The for-loop would do nothing because of the range, correct? Is the length check before a range a common thing to do?